src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -378,7 +378,7 @@ private static void CreateMethod(string propIdentifier, string propType, CodeCla
                 Optional = asOptional,
                 Description = parameter.Description.CleanupDescription(),
                 Kind = CodeParameterKind.Path,
-                SerializationName = parameter.Name.Equals(codeName) ? default : parameter.Name.SanitizeParameterNameForUrlTemplate(),
+                SerializationName = parameter.Name.Equals(codeName) ? codeName : parameter.Name.SanitizeParameterNameForUrlTemplate(),
```

Comment:
I think the `SerializationName` being empty was intentional. As I understand it, if it's empty, read the `Name` instead

Reply:
I don't think its the intention as the writers for the various language seem to not check for it when looking for it. 

https://github.com/microsoft/kiota/blob/0f2c411ebcecc2047d80499ec12276c76ee1f5b4/src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs#L146

https://github.com/microsoft/kiota/blob/0f2c411ebcecc2047d80499ec12276c76ee1f5b4/src/Kiota.Builder/Writers/Java/CodeMethodWriter.cs#L161

https://github.com/microsoft/kiota/blob/0f2c411ebcecc2047d80499ec12276c76ee1f5b4/src/Kiota.Builder/Writers/Go/CodeMethodWriter.cs#L269

Reply:
The code you're referencing selects both the `Name` and `SerializationName`.

Reply:
`SerializationName` refers to the key value and the **transformed** `Name` is the key value passed to the generated function. 

The values are passed to similar equivalent functions as below where the `SerializationName` is the `Item2` instance. If we don't have a value set for it, we will insert empty values across the languages.  

https://github.com/microsoft/kiota/blob/0f2c411ebcecc2047d80499ec12276c76ee1f5b4/src/Kiota.Builder/Writers/CSharp/CSharpConventionService.cs#L42